src/parseTree/nodes/node.java

```diff
@@ -0,0 +1,15 @@
+package src.parseTree.nodes;
+
+public interface node {
+    /**
+     * Adds a child to the node's internal list of children
+     *
+     * @param child The child to add to the node
+     */
+    public void addChild(Object child);
```

Comment:
we should make node an abstract class and just implement the children list, addChild, and getChildren to DRY up all of those redundant definitions